clients/tabby-agent/src/protocol.ts

```diff
@@ -439,8 +439,34 @@ export type ChatEditParams = {
    *    use {@link ChatEditResolveRequest} to resolve it later.
    */
   format: "previewChanges";
+
+  /**
+   * list of file contexts.
+   */
+  context?: ChatEditFileContext[];
 };
 
+/**
+ * Represents a file context use in {@link ChatEditParams}.
+ */
+export interface ChatEditFileContext {
+  /**
+   * The symbol in the user command that refer to this file context.
+   */
+  referer: string;
+
+  /**
+   * The uri of the file.
+   */
+  path: string;
```

Comment:
I perfer to rename it to `uri: URI`
```suggestion
  uri: URI;
```